Support `get_element` from LIST column (#8071)
rapidsai/cudf: cpp/docs/DEVELOPER_GUIDE.md

which are all derived from `cudf::scalar`. e.g. A `numeric_scalar` holds a single numerical value, 
a `string_scalar` holds a single string. The data for the stored value resides in device memory.

> A `list_scalar` holds the underlying data of a list. This means the underlying data can be any type
> that cudf supports. e.g. A `list_scalar` representing a list of integers stores an `cudf::column`
> of type `INT32`, a `list_scalar` represents a list of list of integers stores an `cudf::column` of
> type `LIST`, which in turn stores a column of type `INT32`.

Reviewer: ```suggestion
A `list_scalar` holds the underlying data of a single list. This means the underlying data can be any type
that cudf supports. For example, a `list_scalar` representing a list of integers stores a `cudf::column`
of type `INT32`. A `list_scalar` representing a list of lists of integers stores a `cudf::column` of
type `LIST`, which in turn stores a column of type `INT32`.
```

|timestamp|`timestamp_scalar<T>` | `T` can be `timestamp_D`, `timestamp_s`, etc.|
|duration|`duration_scalar<T>` | `T` can be `duration_D`, `duration_s`, etc.|
|string|`string_scalar`| This class object is immutable|
|list|`list_scalar`| Underlying data can be any type supported by cudf |

### Construction
`scalar`s can be created using either their respective constructors or using factory functions like 
`make_numeric_scalar()`, `make_timestamp_scalar()` or `make_string_scalar()`. 

// TODO: add details for `list_scalar`

> 

Reviewer: Finish these TODOs before merging?

Author: This PR did not introduce any new factory functions for `list_scalar`, aka `make_list_scalar(...)`. I'm not quite sure at this point what should the arguments be for this.

Reviewer: So do we need these TODOs?

Author: The paragraph above suggests a factory function is available for each type of scalars. Maybe instead of adding `TODO` we should rephrase the words above to exclude `list_scalar`.
